### PR TITLE
Performance driver for BVH

### LINKED ISSUE #211: Create a performance driver for BVH
Should allow for:
- [x] running using different node types 
- [x] changing the number of points
- [x] switching between knn and radius searches

Base it on the linearbvh test.

### PR DESCRIPTION
**Pull request by aprokop**
Fix #211

### CODE REVIEW

**Comment by aprokop** [comment]
It's too early to look at the actual code, but I'm already getting segfaults when I try to vary the mesh sizes and running with "radius" option. I am looking into it.

**Comment by aprokop** [comment]
The code is pretty early, and has a lot of overlap with `tstLinearBVH`. Would probably be a good idea to move the shared parts to some common utils things.

File: packages/Kokkos/examples/bvh_driver/bvh_driver.cpp [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    double Lx = 100.0;
+    double Ly = 100.0;
+    double Lz = 100.0;
+    int nx = 11;
+    int ny = 11;
+    int nz = 11;
+    int n_points = 100;
+    std::string mode = "raidus";
```
Comment (Rombur) [inline packages/Kokkos/examples/bvh_driver/bvh_driver.cpp]: typo

**Comment by aprokop** [comment]
Running
`gdb --args ./DataTransferKitKokkos_bvh.exe --node=serial --mode="radius" --nx=100 --node=serial`

results in segfault with backtrace of more than 2115. Sounds like an infinite recursion.

File: packages/Kokkos/examples/bvh_driver/bvh_driver.cpp [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( int const i ) const
+    {
+        DataTransferKit::Details::Within within_predicate(
+            {_point_coords( i, 0 ), _point_coords( i, 1 ),
+             _point_coords( i, 2 )},
+            _radii( i ) );
+        int indices[_max_n_indices];
```
Comment (sslattery) [inline packages/Kokkos/examples/bvh_driver/bvh_driver.cpp]: _max_n_indices is not `static` or `constexpr` so this will not compile (see fetnat buildlog)

**Comment by aprokop** [comment]
Rebased on top of master. Still need to fix @sslattery's comment:
`_max_n_indices is not static or constexpr so this will not compile (see fetnat buildlog)`

**Comment by aprokop** [comment]
Updated PR: not a fix, but rather a workaround.

**Pull request merged**